### PR TITLE
disable network lint

### PR DESCRIPTION
**Pull request by GuyAfik**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-5505)

## Description
add the option to disable docker networks when running unit-tests for integrations/scripts.

## Must have
- [x] Tests

### CODE REVIEW

File: demisto_sdk/commands/lint/linter.py
Comment (dorschw) [inline demisto_sdk/commands/lint/linter.py:193]:
```suggestion
    def should_disable_network(self) -> bool
```

File: demisto_sdk/commands/lint/linter.py
Comment (dorschw) [inline demisto_sdk/commands/lint/linter.py:206]: please use `get_id` here instead of getting it manually

File: demisto_sdk/commands/lint/linter.py
Comment (dorschw) [inline demisto_sdk/commands/lint/linter.py:755]:
```suggestion
            should_disable_network(bool): whether network is not required when running pytest (default: disabled)
```

Comment (dorschw) [inline demisto_sdk/commands/lint/linter.py:755]:
```suggestion
            should_disable_network(bool): whether network is required for linting
```

File: demisto_sdk/commands/lint/linter.py
Comment (dorschw) [inline demisto_sdk/commands/lint/linter.py:1092]: for easier reading, please only show `network: enabled/disabled` (instead of True/False for `disabled`)

File: demisto_sdk/commands/lint/tests/test_linter/gather_facts_test.py
Comment (dorschw) [inline demisto_sdk/commands/lint/tests/test_linter/gather_facts_test.py:724]: please add type hints

File: demisto_sdk/commands/lint/linter.py
Comment (GuyAfik) [reply demisto_sdk/commands/lint/linter.py:206]: if you mean a function, there isn't such one, exists only in `demisto_sdk/commands/common/content/objects/abstract_objects/dictionary_based_object.py`

but its a method that is bounded to an object, in my case its a raw yml file.

Comment (dorschw) [reply demisto_sdk/commands/lint/linter.py:206]: @ilaner maybe we can use the graph for that?

File: demisto_sdk/commands/lint/linter.py
Comment (dorschw) [reply demisto_sdk/commands/lint/linter.py]: are all of them blocked together, or does blocking only happen for pytest?

File: demisto_sdk/commands/lint/linter.py
Comment (GuyAfik) [reply demisto_sdk/commands/lint/linter.py:206]: why would we need to use the graph here?

File: demisto_sdk/commands/lint/linter.py
Comment (dorschw) [reply demisto_sdk/commands/lint/linter.py]: nah, let's keep it pytest only, and i'll resolve this comment. thanks :)

File: demisto_sdk/commands/lint/linter.py
Comment (JudahSchwartz) [inline demisto_sdk/commands/lint/linter.py:1095]:
```suggestion
            f"{self._pack_name} - Pytest - Image {test_image}, network: {network_status}"
```

File: demisto_sdk/commands/lint/linter.py
Comment (JudahSchwartz) [inline demisto_sdk/commands/lint/linter.py:201]: Highly reccomend to refactor this section with the code that does the same thing already in tools and validate_manager.

Comment (GuyAfik) [reply demisto_sdk/commands/lint/linter.py:201]: what do you exactly suggests?

Comment (JudahSchwartz) [reply demisto_sdk/commands/lint/linter.py:201]: we should make a function that takes 

_pack_ignore_file_path = Path(
            f"Packs/{pack_name}/{PACKS_PACK_IGNORE_FILE_NAME}"
        )
        if _pack_ignore_file_path.exists():
            config = ConfigParser(allow_no_value=True)
            config.read(_pack_ignore_file_path)
            if TESTS_REQUIRE_NETWORK_PACK_IGNORE in config.sections():
            
            and just returns sections as a dict so it can be reused

Comment (GuyAfik) [reply demisto_sdk/commands/lint/linter.py:194]: applied your suggested changed

File: demisto_sdk/commands/lint/linter.py
Comment (dorschw) [reply demisto_sdk/commands/lint/linter.py:199]: never mind - but please extract the logic there into an external function (unbound it from the object), possibly move it to `tools.py` and re-use it here. One day I'll make it to implement https://github.com/demisto/demisto-sdk/pull/2345, but until then.. let's not make _more_ duplicates of this logic.

Comment (GuyAfik) [reply demisto_sdk/commands/lint/linter.py:199]: so to get things straight, would you like me to extract `self._yml_file_content.get("commonfields", {}).get("id")` to an external function?

Comment (dorschw) [reply demisto_sdk/commands/lint/linter.py:199]: extract the one at `dictionary_based_object.py` into `tools.py`, and use it both here and there.

Comment (GuyAfik) [reply demisto_sdk/commands/lint/linter.py]: done

File: demisto_sdk/commands/common/tools.py
Comment (JudahSchwartz) [inline demisto_sdk/commands/common/tools.py:1391]: theres a lot of nesting here and it isnt clear what we're doing. Lets split into functions with names that make it clearer

Comment (dorschw) [reply demisto_sdk/commands/common/tools.py:1391]: perhaps using `more_itertools`, or some Sourcery

Comment (GuyAfik) [reply demisto_sdk/commands/common/tools.py:1391]: this code is old, wasn't actually changed by me, so after speaking with @JudahSchwartz ill disregard it

**Review by JudahSchwartz — APPROVED** [review]
lets wait until the feature freeze is over before merge. On merge, lets rerun lint all. 
Nice work!

File: demisto_sdk/commands/common/tools.py
Comment (JudahSchwartz) [inline demisto_sdk/commands/common/tools.py:1352]: can use warn, i think it makes it orange

Comment (GuyAfik) [reply demisto_sdk/commands/common/tools.py:1352]: originally it was `info`, still change it to warning?

**Comment by GuyAfik** [comment]
@dorschw @JudahSchwartz please re-review as its been stale for quite some time.

File: demisto_sdk/commands/common/tests/tools_test.py
Comment (dorschw) [inline demisto_sdk/commands/common/tests/tools_test.py:1008]:
```suggestion
        assert type(get_pack_ignore_content(pack.name)) is type(expected_object)
```

File: demisto_sdk/commands/common/tools.py
Comment (dorschw) [inline demisto_sdk/commands/common/tools.py:1350]:
```suggestion
            logger.exception(
                f"Error when retrieving the content of {_pack_ignore_file_path}\n"
            )
```
logger.exception adds the stack trace :)

File: demisto_sdk/commands/common/tools.py
Comment (dorschw) [reply demisto_sdk/commands/common/tools.py:1352]: yes, imo

File: demisto_sdk/commands/common/tools.py
Comment (dorschw) [inline demisto_sdk/commands/common/tools.py:1387]:
```suggestion
        for section in filter(lambda section: section.startswith("file:"), config.sections():
```
then unindent everything

File: demisto_sdk/commands/common/tools.py
Comment (dorschw) [inline demisto_sdk/commands/common/tools.py:3514]: replace all `if x in y.keys()` with `if x in y` please

File: demisto_sdk/commands/common/tools.py
Comment (dorschw) [inline demisto_sdk/commands/common/tools.py:3527]: these can be replaced with 
```python
for key in ("global_rule_id", "trigger_id", "content_global_id", "rule_id"): 
    if key in file_content: 
        return file_content[key]
```

File: demisto_sdk/commands/lint/linter.py
Comment (dorschw) [inline demisto_sdk/commands/lint/linter.py:210]:
```suggestion
                if get_id(self._yml_file_content) in ignored_integrations_scripts_ids:
                    return False
```

File: demisto_sdk/commands/validate/validate_manager.py
Comment (dorschw) [inline demisto_sdk/commands/validate/validate_manager.py:2670]: don't we have a function that reads the config file this way somewhere, that we can use? 🤔

File: CHANGELOG.md
Comment (JudahSchwartz) [inline CHANGELOG.md:4]: Lets leave a link to the docs

File: demisto_sdk/commands/common/tools.py
Comment (JudahSchwartz) [reply demisto_sdk/commands/common/tools.py:1387]: nice

**Review by JudahSchwartz — APPROVED** [review]
nice

File: CHANGELOG.md
Comment (dorschw) [reply CHANGELOG.md:4]: here?

File: demisto_sdk/commands/common/tools.py
Comment (GuyAfik) [reply demisto_sdk/commands/common/tools.py:1387]: indeed nice, will update

File: demisto_sdk/commands/common/tools.py
Comment (GuyAfik) [reply demisto_sdk/commands/common/tools.py:3527]: done

File: demisto_sdk/commands/validate/validate_manager.py
Comment (GuyAfik) [reply demisto_sdk/commands/validate/validate_manager.py:2670]: as far as i know we don't have, but ill double check.

File: CHANGELOG.md
Comment (JudahSchwartz) [reply CHANGELOG.md:4]: yeah disregard this. not the right place to put it

Comment (dorschw) [inline CHANGELOG.md:4]:
```suggestion
* **lint** now prevents unit-tests from accessing online resources in runtime.
```

File: demisto_sdk/commands/common/tools.py
Comment (dorschw) [inline demisto_sdk/commands/common/tools.py:3537]:
```suggestion
        return file_content["commonfields"].get("id")
```

File: demisto_sdk/commands/common/tools.py
Comment (dorschw) [inline demisto_sdk/commands/common/tools.py:3541]:
```suggestion
        return file_content["dashboards_data"][0].get("global_id")
    elif "templates_data" in file_content:
        return file_content["templates_data"][0].get("global_id")
```

**Pull request merged**
